### PR TITLE
[DOC store] convert api examples to native classes/Octane

### LINKED ISSUE #7256: API docs do not match Octane blueprints
The API docs do not reflect the native classes generated with Octane. I didn't see any open issues on this and didn't know if this is in the works or something that's looking for a volunteer? 

I see some of the example code has been updated in 3.20.

edit - add package checklist

Packages:
- [x] adapter - pr #7286
- [x] model - pr #7292 
- [x] serializer - pr #7296
- [x] store - #7297

### PR DESCRIPTION
**Pull request by maxwondercorn**
Updated store examples - this is last PR to resolve #7256.

I'm going to open a new issue because I realized that we document two different imports for the base adapter but only one is generated with the blueprints.

```javascript
// generated
import ApplicationAdapter from "./application";

// used but not generated
import Adapter from '@ember-data/adapter';
```

### CODE REVIEW

File: packages/store/addon/-private/system/core-store.ts [hunk 1]
```diff
@@ -856,6 +855,7 @@ abstract class CoreStore extends Service {
     // app/adapters/post.js
     import ApplicationAdapter from "./application";
 
+    export default class PostAdapter extends ApplicationAdapter {
     export default ApplicationAdapter.extend({
```
Comment (snewcomer) [inline packages/store/addon/-private/system/core-store.ts:859]: ✂️ ?

File: packages/store/addon/-private/system/core-store.ts [hunk 2]
```diff
@@ -270,12 +269,12 @@ abstract class CoreStore extends Service {
     ```js
     import Store from '@ember-data/store';
 
-    export default Store.extend({
+    export default class MyStore extends Store {
       init() {
         this._super(...arguments);
```
Comment (snewcomer) [inline packages/store/addon/-private/system/core-store.ts:274]: `super.init(...arguments);`

File: packages/store/addon/-private/system/snapshot-record-array.ts [hunk 3]
```diff
@@ -47,11 +47,11 @@ export default class SnapshotRecordArray {
       ```app/adapters/post.js
       import JSONAPIAdapter from '@ember-data/adapter/json-api';
 
-      export default JSONAPIAdapter.extend({
+      export default class PostAdapter extends JSONAPIAdapter {
         shouldReloadAll(store, snapshotRecordArray) {
           return !snapshotRecordArray.length;
         },
```
Comment (snewcomer) [inline packages/store/addon/-private/system/snapshot-record-array.ts:53]: ✂️  `,`

File: packages/store/addon/-private/system/snapshot-record-array.ts [hunk 4]
```diff
@@ -69,13 +69,13 @@ export default class SnapshotRecordArray {
...
 
-      export default JSONAPIAdapter.extend({
+      export default class PostAdapter extends JSONAPIAdapter {
         shouldReloadAll(store, snapshotRecordArray) {
-          var lastRequestTime = snapshotRecordArray.meta.lastRequestTime;
-          var twentyMinutes = 20 * 60 * 1000;
+          let lastRequestTime = snapshotRecordArray.meta.lastRequestTime;
+          let twentyMinutes = 20 * 60 * 1000;
           return Date.now() > lastRequestTime + twentyMinutes;
         },
```
Comment (snewcomer) [inline packages/store/addon/-private/system/snapshot-record-array.ts:77]: ✂️  `,`

**Review by snewcomer — APPROVED** [review]
Phenomenal job!  Readers are going to appreciate this. Just a few comments.

File: packages/store/addon/-private/system/core-store.ts (hunk 2 above)
Comment (maxwondercorn) [reply packages/store/addon/-private/system/core-store.ts:274]: I also missed converting the init() to constructor(). Fixed

File: packages/store/addon/-private/system/snapshot-record-array.ts (hunk 3 above)
Comment (maxwondercorn) [reply packages/store/addon/-private/system/snapshot-record-array.ts:53]: fixed

File: packages/store/addon/-private/system/snapshot-record-array.ts (hunk 4 above)
Comment (maxwondercorn) [reply packages/store/addon/-private/system/snapshot-record-array.ts:77]: fixed

File: packages/store/addon/-private/system/core-store.ts (hunk 1 above)
Comment (maxwondercorn) [reply packages/store/addon/-private/system/core-store.ts:859]: Not sure what needs to be cut/fixed here?

Comment (snewcomer) [reply packages/store/addon/-private/system/core-store.ts:859]: Looks like the class is duplicated...So ✂️ `export default ApplicationAdapter.extend({`

Comment (maxwondercorn) [reply packages/store/addon/-private/system/core-store.ts:859]: I stared at that for five whole  minutes and just couldn't see it :(

File: packages/store/addon/-private/system/core-store.ts [hunk 5]
```diff
@@ -854,9 +853,9 @@ abstract class CoreStore extends Service {
 
     ```js
     // app/adapters/post.js
-    import ApplicationAdapter from "./application";
+    import ApplicationAdapter from - "./application";
```
Comment (sly7-7) [inline packages/store/addon/-private/system/core-store.ts:856]: Is the`-` a typo ?

Comment (maxwondercorn) [reply packages/store/addon/-private/system/core-store.ts:856]: yep, fat fingers - fixed

**Pull request merged**